src/Main.elm

```diff
@@ -143,8 +149,15 @@ type alias FormField =
     }
 
 
+type alias CustomElement =
+    { inputType : String
+    , inputTag : String
+    , attributes : Dict String String
+    }
+
+
 type InputField
-    = ShortText String (List ( String, String ))
+    = ShortText CustomElement
     | LongText (Maybe Int)
```

Comment:
crux of the refactor. `inputTag : String` is the new attribute we are storing.

bulk of the PR is about adapting to this model change